src/batch/src/executor/iceberg_scan.rs

```diff
@@ -42,14 +44,48 @@ use crate::task::BatchTaskContext;
 
 static POSITION_DELETE_FILE_FILE_PATH_INDEX: usize = 0;
 static POSITION_DELETE_FILE_POS: usize = 1;
+
+pub enum IcebergFileScanTaskEnum {
+    DataAndPositionDelete(Vec<FileScanTask>, Vec<FileScanTask>),
```

Comment:
Some docstrings which indicate the following will be helpful here:
1. The semantics.
2. Some reference docs (e.g. iceberg docs on these)